Contextual Onboarding Linear

alessandroboron opened the pull request:
Task/Issue URL: https://app.asana.com/0/1206329551987282/1207812824986102/f

**Description**:

Show the Contextual linear flow. 

**Note**
I will add tests in a subsequent PR. I wanted to unblock you so you can use this branch.

**Steps to test this PR**:
1. Add temporary `newInstallCompletion(self)` in `DefaultVariantManager` at line 135
2. Enable environment variable VARIANT and set value to `mb`
3. Delete App from simulator.
4. Launch App.
5. Complete Onboarding flow

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

alessandroboron committed:
Remove ContextualOnboardinBackgroundWrapper and use View extension

alessandroboron committed:
Make @State private, add action when blocked trackers dialog action is pressed and general clean up

alessandroboron committed:
Implement factory for experiment Contextual Dialogs

alessandroboron committed:
Forward Contextual Onboarding events to main view controller

alessandroboron committed:
Dismiss Contextual Dialog if there are no browsing spec to show

alessandroboron committed:
Refactor Contextual dialog factory to use only one type of communication pattern

alessandroboron committed:
Highlight blocked trackers in dax dialog

alessandroboron committed:
Dismiss Fire button animation when the user taps on it

alessandroboron committed:
Add TODOs

alessandroboron committed:
Fix tests

SabrinaTardio commented on DuckDuckGo/OnboardingExperiment/ContextualDaxDialogs/ContextualOnboardingDialogs.swift:
```diff
@@ -74,12 +75,13 @@ struct OnboardingFireButtonDialog: View {
     let cta = UserText.DaxOnboardingExperiment.ContextualOnboarding.onboardingGotItButton
+
+    @State private var showNextScreen: Bool = false
+    @State private var shouldFollowUp: Bool = true
```
shouldn’t this come from outside? (based on wether or not a website has already been visited?

SabrinaTardio reviewed (commented):
Linear Flow works as expected!
If you visit a site and then do a search, the search contextual appears and after pressing got it still shows the site suggestion.

I left a comment on this: showNextScreen in OnboardingFirstSearchDoneDialog should be set accordingly to whether a site has been visited or not

alessandroboron committed:
Fix PR comments

alessandroboron committed:
Add unit tests for TabViewController, ContextualOnboarding presenter and updated OnboardingNavigationDelegate tests

SabrinaTardio commented on DuckDuckGo/DaxDialogs.swift:
```diff
@@ -317,7 +341,39 @@ final class DaxDialogs: NewTabDialogSpecProvider {
+        guard let host = privacyInfo.domain else { return nil }
+
+        if privacyInfo.url.isDuckDuckGoSearch {
+            // If user already performed a search, show after search dialog but don't follow up by suggesting to visit a site.
```
I guess here should be “If user already visited a site..."

SabrinaTardio commented on DuckDuckGoTests/TabViewControllerDaxDialogTests.swift:
```diff
@@ -0,0 +1,105 @@
+        try super.setUpWithError()
+        delegateMock = MockTabDelegate()
+        onboardingPresenterMock = ContextualOnboardingPresenterMock()
+        sut = .mock(contextualOnboardingPresenter: onboardingPresenterMock)
```
I would not call this tab mock cause it’s a real tab I’d call it fixture? A standard initialisation

alessandroboron replied:
Changed to "fake"

SabrinaTardio reviewed (approved):
Works great now! thanks!
I left 2 very minor comments

SabrinaTardio committed:
improve contextual dialogs (#3093)

Task/Issue URL: https://app.asana.com/0/0/1207812824983911/f
https://app.asana.com/0/0/1207812824983904/f

Description: Makes the Dialogs scrollable and refactors the ContextualDialogs to support expansion animation

alessandroboron committed:
fix pr comments
